packages/sdk/src/NomadContext.ts

```diff
@@ -173,6 +173,43 @@ export class NomadContext extends MultiProvider<config.Domain> {
     return this.mustGetCore(this.governor.domain);
   }
 
+  async process(txId: string): Promise<ContractTransaction>{
+    const isProduction = this.conf.environment === 'production'
+    const nomadAPI = isProduction
+      ? 'https://bridge-indexer.prod.madlads.tools/tx/'
+      : 'https://bridge-indexer.dev.madlads.tools/tx/'
+    const s3URL = isProduction
+      ? 'https://nomadxyz-production-proofs.s3.us-west-2.amazonaws.com/'
+      : 'https://nomadxyz-development-proofs.s3.us-west-2.amazonaws.com/'
+
+    // get transfer message
+    const res = await fetch(`${nomadAPI}${txId}`)
+    const tx = (await res.json())[0] as any
+
+    // get proof
+    const index = BigNumber.from(tx.leafIndex).toNumber()
```

Comment:
i think SDK should query on-chain, not data from indexer. i don't think we have enough load-testing to make indexer stable for external queries. all of the transaction information, though, is available in the `NomadMessage` object in the SDK

Reply:
SDK needs to query indexer eventually. the chain event querying is not sufficient and leads to the terrible kludges we have. Are you arguing that now is not the time?

Reply:
probably not the time, given 
- the indexer is running from a feature branch; it hasn't been finished or merged to main 
- it's been live running in production for a week 
- there is no documentation 
- there is no monitoring / tracing to debug issues in the code
- there is one engineer who has a deep understanding of the code

Reply:
we are working on all of the above FWIW